docs/basics/101-130-yodaproject.rst

```diff
@@ -0,0 +1,564 @@
+   plot.savefig('output/pairwise_relationships.png')
+
+   # perform a K-nearest-neighbours classification with scikit-learn
+   # Step 1: split data in test and training dataset (20:80)
+   array = df.values
+   X = array[:,0:4]
+   Y = array[:,4]
+   test_size = 0.20
+   seed = 7
+   X_train, X_test, Y_train, Y_test = model_selection.train_test_split(X, Y,
+                                                                       test_size=test_size,
+                                                                       random_state=seed)
+   # Step 2: Fit the model and make predictions on the test dataset
+   knn = KNeighborsClassifier()
+   knn.fit(X_train, Y_train)
+   predictions = knn.predict(X_test)
+
+   # Step 3: Save the classification report
+   report = classification_report(Y_test, predictions, output_dict=True)
+   df_report = pd.DataFrame(report).transpose().to_csv('output/prediction_report.csv')
+
+   EOT
+
+This script will
+
+- make sure to install the subdataset (line 11),
+- retrieve the data prior to reading it in (l. 12), and
+- save the resulting figure (l. 21) and ``.csv`` file (l 40) into the ``output/`` directory.
+
+Note how all paths (to input data and output files) are *relative*, such that the
+``midterm_project`` analysis is completely self-contained within the dataset.
+
+Let's run a quick :command:`datalad status`...
+
+.. runrecord:: _examples/DL-101-130-111
+   :language: console
+   :workdir: dl-101/DataLad-101/midterm_project
+
+   $ datalad status
+
+... and save the script to the subdataset's history:
+
+.. runrecord:: _examples/DL-101-130-112
+   :language: console
+   :workdir: dl-101/DataLad-101/midterm_project
+
+   $ datalad save -m "add script for kNN classification and plotting" code/script.py
+
+Finally, with your directory structure being modular and intuitive,
+the input data installed, the script ready, and the dataset status clean,
+you can wrap the execution of the script in a :command:`datalad run` command.
+
+.. note::
+
+   Note that you need to have the following Python packages installed to run the
+   analysis [#f3]_:
+
+   - `pandas <https://pandas.pydata.org/>`_
+   - `seaborn <https://seaborn.pydata.org/>`_
+   - `sklearn <https://scikit-learn.org/>`_
+
+   The packages can be installed via ``pip``. Check the footnote [#f3]_ for code
+   snippets to copy and paste. However, if you do not want to install any
+   Python packages, do not execute the remaining code examples in this sections
+   -- an upcoming section on ``datalad containers-run`` will allow you to
+   perform the analysis without changing with your Python software-setup.
+
+.. runrecord:: _examples/DL-101-130-113
+   :language: console
+   :workdir: dl-101/DataLad-101/midterm_project
+
+   $ datalad run -m "analyze iris data with classification analysis" \
+     --input "input/iris.csv" \
+     --output "output/prediction_report.csv" \
+     --output "output/pairwise_relationships.png" \
+     "python3 code/script.py"
```

Comment:
In principle `code/script.py` is also "input". Only because we know that it is tracked in Git within this very dataset, we can skip that. If we listed it as input, DataLad would also obtain potentially required subdatasets with code (or entire toolboxes) on demand.